pkg/pn/groupInfo_test.go

```diff
@@ -0,0 +1,275 @@
+			Expected: GroupInfo{
+				FamilyName: "CROUCH",
+				GivenName:  "BARTEMIUS",
+				MiddleName: "",
+				NamePrefix: "",
+				NameSuffix: "",
+			},
+		},
+		// No first
+		{
+			Raw: "CROUCH^^'BARTY'^MR^JR",
+			Expected: GroupInfo{
+				FamilyName: "CROUCH",
+				GivenName:  "",
+				MiddleName: "'BARTY'",
+				NamePrefix: "MR",
+				NameSuffix: "JR",
+			},
+		},
+		// Empty
+		{
+			Raw: "^^^^",
+			Expected: GroupInfo{
+				FamilyName: "",
+				GivenName:  "",
+				MiddleName: "",
+				NamePrefix: "",
+				NameSuffix: "",
+			},
+			IsEmpty: true,
+		},
+		// No suffix or trailing
+		{
+			Raw: "CROUCH^BARTEMIUS^'BARTY'^MR",
+			Expected: GroupInfo{
+				FamilyName: "CROUCH",
+				GivenName:  "BARTEMIUS",
+				MiddleName: "'BARTY'",
+				NamePrefix: "MR",
+				NameSuffix: "",
+			},
+			RemoveTrailingSeps: true,
+		},
+		// No prefix or trailing
+		{
+			Raw: "CROUCH^BARTEMIUS^'BARTY'",
+			Expected: GroupInfo{
+				FamilyName: "CROUCH",
+				GivenName:  "BARTEMIUS",
+				MiddleName: "'BARTY'",
+				NamePrefix: "",
+				NameSuffix: "",
+			},
+			RemoveTrailingSeps: true,
+		},
+		// No middle or trailing
+		{
+			Raw: "CROUCH^BARTEMIUS",
+			Expected: GroupInfo{
+				FamilyName: "CROUCH",
+				GivenName:  "BARTEMIUS",
+				MiddleName: "",
+				NamePrefix: "",
+				NameSuffix: "",
+			},
+			RemoveTrailingSeps: true,
+		},
+		// No given or trailing
+		{
+			Raw: "CROUCH",
+			Expected: GroupInfo{
+				FamilyName: "CROUCH",
+				GivenName:  "",
+				MiddleName: "",
+				NamePrefix: "",
+				NameSuffix: "",
+			},
+			RemoveTrailingSeps: true,
+		},
+		// No family or trailing
+		{
+			Raw: "",
+			Expected: GroupInfo{
+				FamilyName: "",
+				GivenName:  "",
+				MiddleName: "",
+				NamePrefix: "",
+				NameSuffix: "",
+			},
+			RemoveTrailingSeps: true,
+			IsEmpty:            true,
+		},
+	}
+
+	for _, thisCase := range testCases {
+		thisCase.Expected.Raw = thisCase.Raw
+
+		// Test creating a new GroupInfo object
+		t.Run(thisCase.Raw+"_New", func(t *testing.T) {
+			newGroup := NewGroupInfo(
+				thisCase.Expected.FamilyName,
+				thisCase.Expected.GivenName,
+				thisCase.Expected.MiddleName,
+				thisCase.Expected.NamePrefix,
+				thisCase.Expected.NameSuffix,
+				thisCase.RemoveTrailingSeps,
+			)
+			assert.Equal(
+				t,
+				thisCase.Raw,
+				newGroup.String(),
+				"convert to string",
+			)
+		})
+
+		// Test parsing a group string.
+		t.Run(thisCase.Raw+"_Parse", func(t *testing.T) {
+			assert := assert.New(t)
+
+			parsed, err := groupFromValueString(thisCase.Raw, "Alphabetic")
+			if !assert.NoError(err, "parse string") {
+				t.FailNow()
+			}
+
+			checkGroupInfo(t, thisCase.Expected, parsed, "")
+
+			assert.Equal(thisCase.Raw, parsed.String(), "convert to string")
+		})
+
+		// Test .IsEmpty() method.
+		t.Run(thisCase.Raw+"_IsEmpty", func(t *testing.T) {
+			assert := assert.New(t)
+
+			parsed, err := groupFromValueString(thisCase.Raw, "Alphabetic")
+			if !assert.NoError(err, "parse string") {
+				t.FailNow()
```

Comment:
Assert will also mark a test as failed and continue on when an assert fails.

t.FailNow() halts test execution and exits when called. I call it here because if an error is returned, all further assertions will fail as well, and continuing will just result in more noise to cut through when reviewing the test logs. If the parse fails, it's not worth checking the parsed value, whatever is causing the error needs to be resolved first.